source/jormungandr/jormungandr/scenarios/ridesharing/klaxit.py

```diff
@@ -0,0 +1,244 @@
+        )
+        self.call_params = None
+
+    def status(self):
+        return {
+            'id': self.system_id,
+            'class': self.__class__.__name__,
+            'circuit_breaker': {
+                'current_state': self.breaker.current_state,
+                'fail_counter': self.breaker.fail_counter,
+                'reset_timeout': self.breaker.reset_timeout,
+            },
+            'network': self.network,
+        }
+
+    def _call_service(self, params):
+        self.logger.debug("requesting Klaxit VIA API")
+
+        headers = {'Authorization': 'apiKey {}'.format(self.api_key)}
+        try:
+            return self.breaker.call(
+                requests.get, url=self.service_url, headers=headers, params=params, timeout=self.timeout
+            )
+        except pybreaker.CircuitBreakerError as e:
+            logging.getLogger(__name__).error(
+                'Klaxit VIA API service dead (error: %s)', e, extra={'ridesharing_service_id': self._get_rs_id()}
+            )
+            raise RidesharingServiceError('circuit breaker open')
+        except requests.Timeout as t:
+            logging.getLogger(__name__).error(
+                'Klaxit VIA API service timeout (error: %s)',
+                t,
+                extra={'ridesharing_service_id': self._get_rs_id()},
+            )
+            raise RidesharingServiceError('timeout')
+        except Exception as e:
+            logging.getLogger(__name__).exception(
+                'Klaxit VIA API service error', extra={'ridesharing_service_id': self._get_rs_id()}
+            )
+            raise RidesharingServiceError(str(e))
+
+    def _make_response(self, raw_json):
+        if not raw_json:
+            return []
+
+        ridesharing_journeys = []
+
+        for offer in raw_json:
+            res = rsj.RidesharingJourney()
+            res.metadata = self.journey_metadata
+            res.distance = offer.get('distance')
+            res.ridesharing_ad = offer.get('webUrl')
+            res.duration = offer.get('duration')
+
+            res.pickup_place = rsj.Place(
+                addr='', lat=offer.get('driverDepartureLat'), lon=offer.get('driverDepartureLng')
+            )
+            res.dropoff_place = rsj.Place(
+                addr='', lat=offer.get('driverArrivalLat'), lon=offer.get('driverArrivalLng')
+            )
+
+            # shape is a list of type_pb2.GeographicalCoord()
+            res.shape = []
+            shape = decode_polyline(offer.get('journeyPolyline'), precision=5)
+            if not shape or res.pickup_place.lon != shape[0][0] or res.pickup_place.lat != shape[0][1]:
+                res.shape.append(type_pb2.GeographicalCoord(lon=res.pickup_place.lon, lat=res.pickup_place.lat))
+
+            res.shape.extend((type_pb2.GeographicalCoord(lon=c[0], lat=c[1]) for c in shape))
+
+            if not shape or res.dropoff_place.lon != shape[0][0] or res.dropoff_place.lat != shape[0][1]:
```

Comment:
```suggestion
            if not shape or res.dropoff_place.lon != shape[-1][0] or res.dropoff_place.lat != shape[-1][1]:
```
You want to test if the **last** element of shape corresponds to the dropoff place right ?

Reply:
I'll do this FIx for blablacar and instant system. Thanks @pbench for the review (3 in 1).
In an other PR, we will extract the code to mutualize it for all ridesharing services